activerecord/lib/active_record/reflection.rb

```diff
@@ -511,13 +511,17 @@ def association_primary_key(klass = nil)
       end
 
       def active_record_primary_key
-        @active_record_primary_key ||= if options[:query_constraints]
+        @active_record_primary_key ||= if options[:primary_key]
```

Comment:
Can you assign options[:primary_key] to a local variable so it's easier to read this method?

Reply:
Done